src/middlewared/middlewared/etc_files/nfsd.py

```diff
@@ -0,0 +1,123 @@
+from collections import defaultdict
+import ipaddress
+import logging
+import os
+
+from middlewared.utils import run
+
+logger = logging.getLogger(__name__)
+
+
+async def get_exports(config, shares, kerberos_keytabs):
+    networks_pool = defaultdict(lambda: defaultdict(lambda: 0))
+
+    result = []
+
+    if config["v4"]:
+        if config["v4_krb"] or kerberos_keytabs:
+            result.append("V4: / -sec=krb5:krb5i:krb5p")
+        else:
+            result.append("V4: / -sec=sys")
+
+    for share in shares:
+        if share["paths"]:
+            share = build_share(config, share, networks_pool)
+            if share:
+                result.append(share)
+
+    return "\n".join(result) + "\n"
+
+
+def build_share(config, share, networks_pool):
+    if share["paths"]:
+        result = list(share["paths"])
+
+        if share["alldirs"]:
+            result.append("-alldirs")
+
+        if share["ro"]:
+            result.append("-ro")
+
+        if share["quiet"]:
+            result.append("-quiet")
+
+        if share["mapall_user"] and share["mapall_group"]:
```

Comment:
This seems to differ from original script, `and` instead of `or`. Intentional?

Reply:
I think this was an error. In django (and now in middlewared) we are validating that if `mapall_user` is present, `mapall_group` must also be present so valid configuration will have either both set or both unset.

Reply:
From the manpage, it seems user can be set without a group